File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
# sliding window // two pointers approach 
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        l = 0
        maxProfit = 0
        for r in range(1,len(prices)):
            if prices[r] > prices[l]:
                maxProfit = max(maxProfit, prices[r] - prices[l])
            else:
                l = r 
        return maxProfit

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.go
func maxProfit(prices []int) int {
    l := 0
	maxProfit := 0
	for r := 1 ; r < len(prices) ; r++ {
		if prices[r] > prices[l] {
			if prices[r] - prices[l] > maxProfit {
				maxProfit = prices[r] - prices[l]
			}
		} else {
			l = r
		}
	}
	return maxProfit

}